driver-core/src/main/java/com/datastax/driver/core/Responses.java

```diff
@@ -82,20 +82,20 @@ private Error(ExceptionCode code, String message, Object infos) {
 
         public DriverException asException(InetSocketAddress host) {
             switch (code) {
-                case SERVER_ERROR:     return new DriverInternalError(String.format("An unexpected error occurred server side on %s: %s", host, message));
-                case PROTOCOL_ERROR:   return new DriverInternalError("An unexpected protocol error occurred. This is a bug in this library, please report: " + message);
+                case SERVER_ERROR:     return new ServerError(host, message);
+                case PROTOCOL_ERROR:   return new ProtocolError(host, message);
```

Comment:
This might be because InvalidQueryException can get created locally (If we can detect the query is invalid), but still think it would be useful in this particular case.